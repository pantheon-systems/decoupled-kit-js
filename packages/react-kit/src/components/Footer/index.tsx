export { Footer } from './Footer';
export * from './props';
