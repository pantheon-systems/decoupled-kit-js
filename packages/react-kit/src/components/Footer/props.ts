export type FooterProps = Readonly<{
	/**
	 * A logo component to render in the footer.
	 * @remarks should be wrapped in a link to the homepage or other relevant page.
	 * @example
	 * ```tsx
	 * export const Logo = ({ exampleLogo }: { exampleLogo: string }) => {
	 * 	// use the link and image component from your framework if applicable.
	 * 	return (
	 * 		<img src={exampleLogo} width="48" height="48" alt="Example Logo" />
	 * 	);
	 * };
	 * ```
	 *
	 */
	Logo:
		| { src: string; alt: string; href: string; styles?: string }
		| React.ReactElement
		| JSX.Element;
	/**
	 * Link used for company website.
	 */
	Link: string;
}>;
