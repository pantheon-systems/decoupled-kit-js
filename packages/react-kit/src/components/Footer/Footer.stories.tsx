import PantheonLogo from '@assets/pantheon-fist-blk.svg';
import { Footer as BottomSignature } from '@components/Footer';
import type { Meta, StoryObj } from '@storybook/react';
import { type FooterProps } from './props';

const Footer = ({ Logo, Link }: Pick<FooterProps, 'Logo' | 'Link'>) => {
	return (
		<div>
			<BottomSignature Logo={Logo} Link={Link} />
		</div>
	);
};

const meta: Meta<typeof Footer> = {
	title: 'Footer/Footer',
	component: Footer,
	parameters: {
		layout: 'fullscreen',
	},
	args: {
		Logo: { src: PantheonLogo, alt: 'Pantheon Systems', href: '/' },
		Link: 'https://pantheon.io/',
	},
	argTypes: {},
	tags: ['autodocs'],
};

export default meta;

type Story = StoryObj<typeof meta>;

export const Default: Story = {};
