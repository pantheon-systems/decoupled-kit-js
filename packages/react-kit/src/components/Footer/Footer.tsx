import React from 'react';
import { FooterProps } from './props';

/**
 * @see {@link https://live-storybook-react-kit.appa.pantheon.site/?path=/docs/footer-footer--docs}
 */
export const Footer = ({ Logo, Link }: FooterProps) => {
	const FooterLogo = () => {
		if (React.isValidElement(Logo)) {
			return Logo;
		} else if (typeof Logo === 'object' && 'src' in Logo && 'alt' in Logo) {
			const { src, alt } = Logo;
			return <img src={src} height="40" alt={alt} className="rk-inline" />;
		}
		return null;
	};
	return (
		<footer className="rk-mt-16 rk-bg-neutral-800 rk-py-4 rk-text-center rk-text-white">
			<div className="rk-mb-0.875 rk-mx-auto rk-mt-2 rk-block">
				<FooterLogo />
			</div>

			<h6 className="rk-mb-0.875 rk-text-lg rk-font-bold">
				Pantheon Decoupled Kit
			</h6>

			<p className="rk-mb-0.875 rk-text-sm">Â© Pantheon 2013</p>

			<p className="rk-mb-2 rk-text-xs">
				Built with{' '}
				<a href={Link} className={'rk-underline'}>
					Pantheon.io
				</a>
			</p>
		</footer>
	);
};
