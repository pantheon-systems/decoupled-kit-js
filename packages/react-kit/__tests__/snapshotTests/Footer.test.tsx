import { Footer } from '@components/Footer';
import { render } from '@testing-library/react';

const Logo = () => {
	return <span>ExampleLogo</span>;
};

const TestFooter = () => {
	return <Footer Logo={Logo} Link="www.test.com"></Footer>;
};

describe('<Footer />', () => {
	it('should render Footer', () => {
		const { asFragment } = render(<TestFooter />);

		expect(asFragment()).toMatchSnapshot();
	});
});
