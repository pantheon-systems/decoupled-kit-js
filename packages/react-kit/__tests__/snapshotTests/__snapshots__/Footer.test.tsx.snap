// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Footer /> > should render Footer 1`] = `
<DocumentFragment>
  <footer
    class="rk-mt-16 rk-bg-neutral-800 rk-py-4 rk-text-center rk-text-white"
  >
    <div
      class="rk-mb-0.875 rk-mx-auto rk-mt-2 rk-block"
    />
    <h6
      class="rk-mb-0.875 font-bold rk-text-lg"
    >
      Pantheon Decoupled Kit
    </h6>
    <p
      class="rk-mb-0.875 rk-text-sm"
    >
      Â© Pantheon 2013
    </p>
    <p
      class="rk-mb-2 rk-text-xs"
    >
      Built with 
      <a
        class="rk-underline"
        href="www.test.com"
      >
        Pantheon.io
      </a>
    </p>
  </footer>
</DocumentFragment>
`;
